front/lib/api/assistant/agent.ts

```diff
@@ -742,25 +743,18 @@ export async function* runMultiActionsAgent(
     });
   }
 
-  yield* tokenEmitter.flushTokens();
+  yield* contentParser.flushTokens();
 
-  const chainOfThought = tokenEmitter.getChainOfThought();
-  const content = tokenEmitter.getContent();
+  const chainOfThought = contentParser.getChainOfThought();
 
-  if (chainOfThought?.length || content?.length) {
+  if (chainOfThought?.length) {
     yield {
       type: "agent_chain_of_thought",
       created: Date.now(),
       configurationId: agentConfiguration.sId,
       messageId: agentMessage.sId,
       message: agentMessage,
-
-      // All content here was generated before a tool use and is not proper generation content
-      // and can therefore be safely assumed to be reflection from the model before using a tool.
-      // In practice, we should never have both chainOfThought and content.
-      // It is not completely impossible that eg Anthropic decides to emit part of the
-      // CoT between `<thinking>` XML tags and the rest outside of any tag.
-      chainOfThought: removeNulls([chainOfThought, content]).join("\n"),
+      chainOfThought,
```

Comment:
no longer including pre-tools-use content in the CoT. CoT is just the content between CoT delimiters.